runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3179,6 +3184,52 @@ remoteCompile(
          JITServer::ServerMemoryState nextMemoryState = std::get<9>(recv);
          updateCompThreadActivationPolicy(compInfoPT, nextMemoryState);
          }
+      else if (JITServer::MessageType::compilationThreadCrashed == response)
+         {
+         // A server compilation thread has crashed, temporarily stop compilations
+         // and request a JitDump recompilation of the method that caused the crash
+         if (TR::Options::isAnyVerboseOptionSet(TR_VerboseJITServer, TR_VerboseCompilationDispatch))
+             TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,
+               "Server compilation thread crashed while compiling %s (%s)", compiler->signature(), compiler->getHotnessName());
+         compInfo->acquireCompMonitor(vmThread);
+         compInfo->getPersistentInfo()->setDisableFurtherCompilation(true);
+         compInfo->purgeMethodQueue(compilationFailure);
+         compInfo->releaseCompMonitor(vmThread);
+         
+         auto recv = client->getRecvData<TR_OptimizationPlan, TR::Options *>();
+         TR_OptimizationPlan jitDumpPlan = std::get<0>(recv);
+         TR::Options *optionsFromOriginalCompile = std::get<1>(recv); 
+
+         J9::JitDumpMethodDetails jitDumpDetails(method, optionsFromOriginalCompile, useAotCompilation);
```

Comment:
The reason why `optionsFromOriginalCompile` exists is because it's used to keep track of the timing-sensitive options:
https://github.com/eclipse/openj9/blob/7822324b051ca16a677382b4f14dae22b2de7473/runtime/compiler/control/CompilationThread.cpp#L8683-L8688.
But you are right that from the client's point of view, the original compilation hasn't ended yet, so the options wouldn't actually change. I guess we could just set `optionsFromOriginalCompile` to `NULL`, to ignore the above section of the code.